TTCN3_Project/CANFilter/com/testingtech/playits/canfilter/Car2XEntryUpdater.java

```
@@ -0,0 +1,139 @@
+          .valueOf((char) 0));
+
+      while (car2xEntries.size() > 0) {
+        parseString(scanner.next());
+      }
+    } catch (SocketTimeoutException e) {
+      System.out.println("[EntryUpdater] Timeout: Could not connect to "
+          + address.getHostName() + ":" + address.getPort());
+    } catch (IOException ioe) {
+      System.err.println("[EntryUpdater] Could not connect to "
+          + address.getHostName() + " on port " + address.getPort());
+    } finally {
+      System.out.println("[EntryUpdater] No more entries to update. Shutting down.");
+      close(scanner);
+      close(socket);
+    }
+  }
+
+  private void close(Scanner scanner) {
+    if (scanner != null) {
+      scanner.close();
+    }
+  }
+
+  private void close(Socket socket) {
+    if (socket != null) {
+      try {
+        socket.close();
+      } catch (IOException e) {
+        System.out.println("[EntryUpdater] Could not close socket: "
+            + e.getMessage());
+      }
+    }
+  }
+
+  private void parseString(String jsonString) {
+//    TODO enable to debug
+//    System.out.println("[EntryUpdater] Incoming object: " + jsonString); 
+    try {
+      JSONObject jsonObject = new JSONObject(jsonString);
+      Car2XEntry car2xEntry = car2xEntries.get(jsonObject.get("name"));
+      if (car2xEntry != null) {
+        updateEntry(car2xEntry, jsonObject);
+      }
+    } catch (JSONException jsone) {
+      System.err.println("[EntryUpdater] Could not parse json string: "
+          + jsonString);
+    }
+  }
+
+  // TODO add obd2 key
+  private void updateEntry(Car2XEntry car2xEntry, JSONObject jsonObject)
+      throws JSONException {
+    car2xEntry.setTimestamp(new Date().getTime());
+    car2xEntry.setValueA(jsonObject.get("value").toString());
+    try {
+      String event = jsonObject.get("event").toString();
+      car2xEntry.setValueB(event);
+    } catch (JSONException jsonException) {
+      // ignore missing event
```

You shouldn't just ignore an Exception.